### DISCUSSION TITLE
One input validates during onBlur the rest of the fields validates on onSubmit

### DISCUSSION THREAD

**Discussion by FilipBrannlund** [react-hook-form/react-hook-form#5362] [Q&A] [answered]
Hello!
First of all, thanks for an amazing package that really speeds up the development process.

I'm having problem when dealing with nested inputs and I'm not sure if RHF och Yup is the place to ask the question.
Validation of the fields with yup is working but RHF is validating different depending on that yup rules I use. 
 
The field labled with "Test" is getting validated onBlur but firstname and lastname is getting validated only when hitting the submit-button.

I want all fields to be validated onBlur and the result of validation should contain in formState.errors

Is the problem related to the validationschema which is the only thing that differs between test and firstname/lastname?

https://codesandbox.io/s/custom-hook-with-resolver-v7-forked-fq44e?file=/src/App.js:886-1172

My CSB is forked from:
https://react-hook-form.com/advanced-usage#CustomHookwithResolver

**Comment by thanh-nguyen-95** [comment, marked as answer]
@bluebill1049 You might want to have a look at this one.

My initial thought was that @FilipBrannlund was using an outdated `react-hook-form` (7.0.0.beta.14). I tried upgrading it to the latest version but the problem still persists.

After some testing, changing the field name to`person[0].firstName` instead of `person.0.firstName` works. I don't usually do this in my projects (I use `useFieldArray` for this case) so I'm not sure if it's a bug or it's not supported.

**Reply by FilipBrannlund** [reply]
@thanh-nguyen-95  Thanks for your answer, I have been into using useFieldArray but I have not found any example on how to iterate out fields from an prefilled object.

For example I have an JSON object containing:
`
{
  "persons": [
    {
      "firstName": "John",
      "lastName": "Doe"
    },
    {
      "firstName": "Jane",
      "lastName": "Doe"
    }
  ]
}
`

And I would like to display the values in input's and make the user edit the values before submitting the form.
Like this:
![test](https://user-images.githubusercontent.com/80246076/119852522-a941ab80-bf0f-11eb-8fe8-7ee8c9aeec84.png)

All the examples I have seen there are always starting with an empty object and then appending new items.

**Reply by bluebill1049** [reply]
>After some testing, changing the field name toperson[0].firstName instead of person.0.firstName works. I don't usually do this in my projects (I use useFieldArray for this case) so I'm not sure if it's a bug or it's not supported.

thanks @thanh-nguyen-95 

we are strictly required to use . syntax for consistent between typescript and javascript. so the bracket is no longer valid for consistency reasons.

**Reply by thanh-nguyen-95** [reply]
> 
> 
> > After some testing, changing the field name toperson[0].firstName instead of person.0.firstName works. I don't usually do this in my projects (I use useFieldArray for this case) so I'm not sure if it's a bug or it's not supported.
> 
> thanks @thanh-nguyen-95
> 
> we are strictly required to use . syntax for consistent between typescript and javascript. so the bracket is no longer valid for consistency reasons.

That's why I said you should have a look. _Bracket syntax_ works instead of _Dot syntax_. I tested on the latest version

Field isn't validated on `onBlur` event when you use _dot syntax_ but works fine when using _bracket syntax_. Shouldn't it be the other way around (_dot syntax_ works and _bracket syntax_ doesn't)

CSB: https://codesandbox.io/s/custom-hook-with-resolver-v7-forked-3v9bp?file=/src/App.js

**Reply by bluebill1049** [reply]
Oh! Interesting. I will double check on this. Thanks 👍

**Reply by thanh-nguyen-95** [reply]
> 
> 
> @thanh-nguyen-95 Thanks for your answer, I have been into using useFieldArray but I have not found any example on how to iterate out fields from an prefilled object.
> 
> For example I have an JSON object containing:
> `{ "persons": [ { "firstName": "John", "lastName": "Doe" }, { "firstName": "Jane", "lastName": "Doe" } ] }`
> 
> And I would like to display the values in input's and make the user edit the values before submitting the form.
> Like this:
> ![test](https://user-images.githubusercontent.com/80246076/119852522-a941ab80-bf0f-11eb-8fe8-7ee8c9aeec84.png)
> 
> All the examples I have seen there are always starting with an empty object and then appending new items.

Passing the default value to `useForm` like below should set the default items for your field array.

```jsx
const methods = useForm({
  mode: 'onBlur',
  resolver: resolver,
  submitFocusError: false,
  defaultValues: {
    people: [
      {
        firstName: 'John',
        lastName: 'Doe'
      },
      {
        firstName: 'Jane',
        lastName: 'Doe'
      }
    ],
    test: 42
  }
})
```

**Reply by bluebill1049** [reply]
raised an issue to track this: https://github.com/react-hook-form/react-hook-form/issues/5377

**Reply by bluebill1049** [reply]
<img width="1100" alt="Screen Shot 2021-05-28 at 1 15 44 pm" src="https://user-images.githubusercontent.com/10513364/119924388-d1d0b080-bfb6-11eb-864c-fae9b01d147e.png">

I can see the correct data is supplied and so does the errors object.

**Reply by bluebill1049** [reply]
<img width="1101" alt="Screen Shot 2021-05-28 at 1 17 54 pm" src="https://user-images.githubusercontent.com/10513364/119924537-1d835a00-bfb7-11eb-885e-79d55ae3260a.png">

looks like it's an issue with the map, it's map flat, instead of nested object.

**Reply by FilipBrannlund** [reply]
@bluebill1049 I can see that the issue has been closed. What is the answer for this question?
Should I use bracket syntax instead of dot syntax?

**Reply by bluebill1049** [reply]
there is no bug in the library, but you should use dot for v7.
